iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/twin/TwinPnPTests.java

```diff
@@ -0,0 +1,185 @@
+package tests.integration.com.microsoft.azure.sdk.iot.iothub.twin;
+
+import com.microsoft.azure.sdk.iot.device.*;
+import com.microsoft.azure.sdk.iot.device.exceptions.ModuleClientException;
+import com.microsoft.azure.sdk.iot.service.RegistryManager;
+import com.microsoft.azure.sdk.iot.service.devicetwin.DeviceTwin;
+import com.microsoft.azure.sdk.iot.service.devicetwin.DeviceTwinDevice;
+import com.microsoft.azure.sdk.iot.service.exceptions.IotHubException;
+import com.microsoft.azure.sdk.iot.service.Device;
+import com.microsoft.azure.sdk.iot.service.Module;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import tests.integration.com.microsoft.azure.sdk.iot.helpers.DeviceConnectionString;
+import tests.integration.com.microsoft.azure.sdk.iot.helpers.IntegrationTest;
+import tests.integration.com.microsoft.azure.sdk.iot.helpers.TestConstants;
+import tests.integration.com.microsoft.azure.sdk.iot.helpers.Tools;
+import tests.integration.com.microsoft.azure.sdk.iot.helpers.annotations.IotHubTest;
+import tests.integration.com.microsoft.azure.sdk.iot.helpers.annotations.StandardTierHubOnlyTest;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.*;
+
+import static com.microsoft.azure.sdk.iot.device.IotHubClientProtocol.*;
+import static com.microsoft.azure.sdk.iot.device.IotHubClientProtocol.MQTT_WS;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test class containing all tests to be run for PnP.
+ */
+@IotHubTest
+@RunWith(Parameterized.class)
+public class TwinPnPTests extends IntegrationTest
+{
+    protected static String iotHubConnectionString = "";
+    private static RegistryManager registryManager;
+    private static final String ModelId = "dtmi:com:test:e2e;1";
+
```

Comment:
Can model id come from constant?

Reply:
sorry didn't get your question.

Reply:
I'd actually like to see this randomized per test so there is no chance of one test interfering with another

Reply:
Is there a restriction to the version number that would keep you from randomizing?

Reply:
Not sure of restriction but added randomizing